frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts

```diff
@@ -127,6 +172,172 @@ export class ExperimentDesignStepperService {
+          weight: '0.0',
+          include: true,
+        };
+
+        tableData.push(tableRow);
+      });
+    });
+
+    return tableData;
+  }
+
+  mergeExistingConditionsTableData(experimentInfo: ExperimentVM): FactorialConditionTableRowData[] {
+    const existingConditions = experimentInfo.conditions;
+    const existingConditionAliases = experimentInfo.conditionAliases;
+    const existingPartitions = experimentInfo.partitions;
+
+    const levelOrder = {};
+    existingPartitions.forEach((partition) => {
+      partition.factors.forEach((factor) => {
+        factor.levels.forEach((level) => {
+          levelOrder[level.id] = factor.order;
+        });
+      });
+    });
+
+    const tableData = existingConditions.map((factorialCondition) => {
+      const conditionAlias = existingConditionAliases.find(
+        (conditionAlias) => conditionAlias?.parentCondition.id === factorialCondition.id
+      );
+
+      const aliasname = conditionAlias ? conditionAlias.aliasName : '';
+      const existingConditionAliasId = conditionAlias?.id;
+
+      factorialCondition.levelCombinationElements.sort((a, b) =>
+        levelOrder[a.level?.id] > levelOrder[b.level?.id]
+          ? 1
+          : levelOrder[b.level?.id] > levelOrder[a.level?.id]
+          ? -1
+          : 0
+      );
+
+      const tableRow: FactorialConditionTableRowData = {
+        id: factorialCondition.id,
+        conditionAliasId: existingConditionAliasId,
+        levels: factorialCondition.levelCombinationElements.map((levelElement) => {
+          return {
+            id: levelElement.level.id,
+            name: levelElement.level.name,
+          };
+        }),
+
+        alias: aliasname,
+        weight: factorialCondition.assignmentWeight.toString(),
+        include: factorialCondition.assignmentWeight > 0,
+      };
+      return tableRow;
+    });
+    return tableData;
+  }
+
+  createFactorialConditionRequestObject() {
+    const tableData = this.getFactorialConditionTableData();
+    const factorialConditionsRequestObject: FactorialConditionRequestObject[] = [];
+    let conditionIndex = 1;
+    tableData.forEach((factorialConditionTableRow) => {
+      factorialConditionsRequestObject.push({
+        id: factorialConditionTableRow.id,
+        name: 'condition ' + conditionIndex, // what should this be?
+        conditionCode: 'condition ' + conditionIndex, // what should this be?
+        assignmentWeight: parseFloat(factorialConditionTableRow.weight),
+        order: conditionIndex++,
+        levelCombinationElements: factorialConditionTableRow.levels.map((level) => {
+          return {
+            id: uuidv4(),
+            level: level,
+          };
+        }),
+      });
+    });
+
+    return factorialConditionsRequestObject;
+  }
+
+  checkConditionTableValidity() {
+    const tableData = this.getFactorialConditionTableData();
+    let sumOfAssignmentWeights = 0.0;
+    let negativeweightError = false;
+
+    if (tableData.length > 0) {
```

Comment:
there has to be a simpler way to do this.

1. negative number check can probably be eliminated before we get to this point by using native html number validation (min = 0)

2. this can probably be accomplished without any type-casting between numbers and strings with a simple `reduce` function.

3. why-oh-why use a ternary for this:
`parseFloat(Number(factorialConditionTableRow.weight).toFixed(1)) < 0 ? (negativeweightError = true) : null;`

Reply:
1. it's not number so min=0 won't work here and it will give that side button if we make it number so let's just keep it that way.
![image](https://user-images.githubusercontent.com/50392803/213629276-f6e6e03e-6269-4840-bfcf-2895ea684a37.png)

2. reduce func will do same thing this is doing, plus we have to convert string to number in that so this seems more optimal in my view.

Reply:
css can remove the number input controls, we have code in `experiment-design.scss` to do exactly this. 

```
 /* Remove arrows/spinners from input type number */
  input::-webkit-outer-spin-button,
  input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  input[type='number'] {
    -moz-appearance: textfield;
  }
}

if we can hide those controls, I'm not sure what benefit there is to writing out a routine native validation manually.

I see that there's already code in the component that is checking for valid weights also, so why is it reimplemented here?

Reply:
min=0 isn't working, it will just limit side controls to 0 but u can type negative values.
In component we are checking for error message in condition table but if person ignore warning or don't go to condition table and move next then this validation will stop it from moving forward as we can't stop them from condition table func so this is added.

Reply:
`Validators.min(0)` would be used in conjunction with this, but we have abandoned Angular's built-in validators for some reason and decided to code everything manually, so I guess we'll code it manually. (see `experiment-form-validators.ts` for an example of the normal way for handling validation in Angular)